extensions/data-transfer/portability-data-transfer-daybook/src/main/java/org/datatransferproject/transfer/daybook/photos/DaybookPhotosImporter.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    // Import photos
+    for (PhotoModel photo : resource.getPhotos()) {
+      executor.executeAndSwallowIOExceptions(
+          photo.getDataId(),
+          photo.getTitle(),
+          () -> {
+            String albumId;
+            if (Strings.isNullOrEmpty(photo.getAlbumId())) {
+              albumId = null;
+            } else {
+              albumId = executor.getCachedValue(photo.getAlbumId());
+            }
+            return importPhoto(photo, jobId, authData, albumId);
+          });
+    }
+
+    // API to trigger daybook API
+
+    return new ImportResult(ImportResult.ResultType.OK);
+  }
+
+  private String importAlbum(PhotoAlbum album, TokensAndUrlAuthData authData) throws IOException {
+    String description = album.getDescription();
+    String album_name = album.getName();
+    monitor.debug(() -> String.format("Album Name: %s", album_name));
+    if (!Strings.isNullOrEmpty(description)) {
+      monitor.debug(() -> String.format("Album description: %s", description));
+    }
+
+    return album_name;
+  }
+
+  private int importPhoto(
+      PhotoModel photoModel, UUID jobId, TokensAndUrlAuthData authData, String newAlbumId)
+      throws IOException {
+    InputStream inputStream = null;
+    String albumId = photoModel.getAlbumId();
+    String imageDescription = photoModel.getDescription();
+    String title = photoModel.getTitle();
+
+    if (photoModel.isInTempStore()) {
+      inputStream = jobStore.getStream(jobId, photoModel.getFetchableUrl()).getStream();
+    } else if (photoModel.getFetchableUrl() != null) {
+      inputStream = new URL(photoModel.getFetchableUrl()).openStream();
+    } else {
+      monitor.severe(() -> "Can't get inputStream for a photo");
+      return -1;
+    }
+
+    byte[] imageBytes = ByteStreams.toByteArray(inputStream);
+    String imageData = Base64.getEncoder().encodeToString(imageBytes);
+
+    Request.Builder requestBuilder = new Request.Builder().url(baseUrl);
+    requestBuilder.header("token", authData.getAccessToken());
+
+    FormBody.Builder builder = new FormBody.Builder().add("image", imageData);
+    builder.add("exporter", JobMetadata.getExportService());
+
+    if (!Strings.isNullOrEmpty(newAlbumId)) {
+      builder.add("album", newAlbumId);
+    }
+
+    if (!Strings.isNullOrEmpty(title)) {
+      builder.add("title", title);
+    }
+
+    if (!Strings.isNullOrEmpty(imageDescription)) {
+      builder.add("description", imageDescription);
+    }
+    FormBody formBody = builder.build();
+    requestBuilder.post(formBody);
+
+    Response response = client.newCall(requestBuilder.build()).execute();
+    int code = response.code();
+    // Though sometimes it returns error code for success requests
+    Preconditions.checkArgument(
+        code >= 200 && code <= 299,
+        String.format(
+            "Error occurred in request for %s, code: %s, message: %s",
+            baseUrl, code, response.message()));
+    return response.code();
```

Comment:
It would be good to enclose this response object in a try-with-resources block:

```suggestion
    try (Response response = client.newCall(requestBuilder.build()).execute()) {
      int code = response.code();
      // Though sometimes it returns error code for success requests
      Preconditions.checkArgument(
          code >= 200 && code <= 299,
          String.format(
              "Error occurred in request for %s, code: %s, message: %s",
              baseUrl, code, response.message()));

      if (photoModel.isInTempStore()) {
        jobStore.removeData(jobId, photoModel.getFetchableUrl());
      }
      return response.code();
    }
```